File: Python/backup_queue.py
import sys
from collections import deque

# 큐 구현하기: 10845번
# queueList = []
# for _ in range(int(sys.stdin.readline())):
#     inputList = sys.stdin.readline().split(" ")
#
#     if inputList[0] == 'push': queueList.append(inputList[1][:-1])
#     if inputList[0][:-1] == 'front':
#         if len(queueList) == 0: print('-1')
#         else: print(queueList[0])
#     if inputList[0][:-1] == 'back':
#         if len(queueList) == 0: print('-1')
#         else: print(queueList[len(queueList) - 1])
#     if inputList[0][:-1] == 'size': print(len(queueList))
#     if inputList[0][:-1] == 'empty':
#         if len(queueList) == 0: print('1')
#         else: print('0')
#     if inputList[0][:-1] == 'pop':
#         if len(queueList) == 0: print('-1')
#         else:
#             print(queueList[0])
#             queueList = queueList[1:]

# 조세퍼스 문제: 1158번
# result_list = []
#
# n, m = map(int, sys.stdin.readline().split())
#
# queue_number = deque([i for i in range(1, n + 1)])
#
# while len(queue_number):
#     for _ in range(m - 1):
#         temp = queue_number.popleft()
#         queue_number.append(temp)
#     result_list.append(str(queue_number.popleft()))
#
# print("<", ", ".join(result_list), ">", sep="")

File: CPP_STUDY/CPP_STUDY/BackupQueue.cpp
int main3() {
    // 큐 구현하기: 10845번
 /* int rangeNumber;

    cin >> rangeNumber;
    int queueList[10000];
    int top = 0; int bottom = 0;

    while (rangeNumber--) {
        string queueMethod;
        cin >> queueMethod;

        if (queueMethod == "push") {
            int inputNumber;
            cin >> inputNumber;
            queueList[bottom] = inputNumber;
            bottom += 1;
        }
        if (queueMethod == "front") {
            if (top == bottom) printf("%d\n", -1);
            else printf("%d\n", queueList[top]);
        }
        if (queueMethod == "back") {
            if (top == bottom) printf("%d\n", -1);
            else printf("%d\n", queueList[bottom - 1]);
        }
        if (queueMethod == "size") printf("%d\n", bottom - top);
        if (queueMethod == "empty") {
            if (top == bottom) printf("%d\n", 1);
            else printf("%d\n", 0);
        }
        if (queueMethod == "pop") {
            if (top == bottom) printf("%d\n", -1);
            else {
                printf("%d\n", queueList[top]);
                top += 1;
            }
        }*/
    
    // 조세퍼스 문제: 1158번
    //int n, m;

    //scanf_s("%d %d", &n, &m);

    //queue<int> queueNumber;

    //for (int i = 1; i < n + 1; i++) queueNumber.push(i);

    //int resultList[5000];
    //int cursor = 0;

    //while (!queueNumber.empty()) {
    //    int temp;
    //    for (int i = 0; i < m - 1; i++) {
    //        temp = queueNumber.front();
    //        queueNumber.pop();
    //        queueNumber.push(temp);
    //    }
    //    resultList[cursor] = queueNumber.front();
    //    queueNumber.pop();
    //    cursor += 1;
    //}

    //printf("<");
    //for (int i = 0; i < cursor - 1; i++) printf("%d, ", resultList[i]);
    //printf("%d", resultList[cursor - 1]);
    //printf(">");

	return 0;
}